### PR TITLE
fix logic for post weight pop up

### PR DESCRIPTION
**Pull request by micahwoodard**
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
- better logic for determining if a session has been run and if the weight of mouse has still not been entered
### What issues or discussions does this update address?
[#615](https://github.com/AllenNeuralDynamics/aind-behavior-blog/issues/615)
### Was this update tested in 446/447?
- [ ] 446

### CODE REVIEW

**Comment by micahwoodard** [comment]
@alexpiet  It seems like after you hit save , the UI is assuming the [next start click will be a new session](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/blob/54d502e89323458a95297803dd3111d37041a666/src/foraging_gui/Foraging.py#L2501C13-L2507C39)? Is that right, and if so, should we check if the post weight has been entered in the case start is pressed after save?

**Comment by alexpiet** [comment]
@micahwoodard This is one of the places where the logic in the code is very messy.

After `_Save()` has run with `SaveContinue==0`, if a user presses `Start` then a new session is started. 

I think we want to check for post weight in the case where (1) trials have been run (this session wasn't loaded from file), and (2) the user presses close/quit, or load (to load a new mouse), or Start (to start a new session).

File: src/foraging_gui/Foraging.py [hunk 1]
```diff
@@ -3103,13 +3105,19 @@ def _Clear(self):
         if self.unsaved_data:
             reply = QMessageBox.critical(self,
                 'Box {}, Clear parameters:'.format(self.box_letter),
-                'Unsaved data exists! Do you want to clear training parameters?',
+                'Unsaved data exists!  n `  training parameters?',
```
Comment (alexpiet) [inline src/foraging_gui/Foraging.py:3108]: What is going on here with line 3108?

**Review by alexpiet — APPROVED** [review]
I think you accidentally deleted one message on line 3108, but looks good other than that

**Comment by micahwoodard** [comment]
I've made changes for behavior when loading in a mouse.

**Review by alexpiet — APPROVED** [review]

**Pull request merged**